core/unpledge/unpledge.go

```diff
@@ -0,0 +1,305 @@
+package unpledge
+
+import (
+	"bufio"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/EnsurityTechnologies/logger"
+	"github.com/rubixchain/rubixgoplatform/core/wallet"
+
+	"golang.org/x/crypto/sha3"
+)
+
+type UnPledge struct {
+	log logger.Logger
+}
+
+func (un *UnPledge) GetPledgedTokenID(did string) ([]string, error) {
+	w := wallet.Wallet{}
+	var tokenIDs []string
+	pledgedTokens, err := w.GetAllPledgedTokens(did)
+	if err != nil {
+		return nil, err
+	}
+	for _, token := range pledgedTokens {
+		tokenIDs = append(tokenIDs, token.TokenID)
+	}
+	return tokenIDs, nil
+}
+
+func (un *UnPledge) GetLastBlockReceiverDID(TokenID string) (string, error) {
+	w := wallet.Wallet{}
+	lastBlock := w.GetLatestTokenBlock(TokenID)
+	if lastBlock == nil {
+		un.log.Error("latest block for token %s not found", TokenID)
+		return "", nil
+	}
+	ReceiverDID := lastBlock.GetReceiverDID()
+	return ReceiverDID, nil
+}
+
+func (un *UnPledge) GetLastBlockTransactionID(TokenID string) (string, error) {
+	w := wallet.Wallet{}
+	lastBlock := w.GetLatestTokenBlock(TokenID)
+	if lastBlock == nil {
+		un.log.Error("latest block for token %s not found", TokenID)
+		return "", nil
+	}
+	TransactionID := lastBlock.GetTid()
+	return TransactionID, nil
+}
+
+func (un *UnPledge) Concat(TokenID string, ReceiverDID string) (string, error) {
+	if TokenID == "" {
+		err := errors.New("TokenID cannot be empty")
+		un.log.Error(err.Error())
+		return "", err
+	}
+	if ReceiverDID == "" {
+		err := errors.New("ReceiverDID cannot be empty")
+		un.log.Error(err.Error())
+		return "", err
+	}
+	valueConcated := TokenID + ReceiverDID
+	return valueConcated, nil
+}
+
+func (un *UnPledge) CalcSHA_256Hash(input string) string {
```

Comment:
looks like these functions are better in a separate class. Do we have a util class or something like that?

for - CalcSHA_256Hash, CalcSHA_3_256Hash, CalcSHA3_256Hash1000Times